bitacross-worker/bitacross/core/bc-musig2-event/src/lib.rs

```diff
@@ -206,25 +221,75 @@ pub fn process_event<OCallApi, SIGNINGAK, SHIELDAK, Responder>(
 				);
 
 				let signer_id = *signer_id;
-				let client = peers_map.lock().unwrap().get(&signer_id).cloned();
-				if let Some(mut client) = client {
-					let request = request.clone();
-					event_threads_pool.execute(move || {
-						if let Err(e) = client.send(&request) {
-							error!(
-								"Could not send request to signer: {:?}, reason: {:?}",
-								signer_id, e
-							)
-						}
-					});
-				} else {
-					error!("Fail to share killing info, unknown signer: {:?}", signer_id);
-				}
+				let peers_map_clone = peers_map.clone();
+				let request = request.clone();
+				let enclave_lookup_cloned = enclave_registry_lookup.clone();
+				let ceremony_registry_cloned = ceremony_registry.clone();
+				let ceremony_id_cloned = ceremony_id.clone();
+				event_threads_pool.execute(move || {
+					send_request(
+						signer_id,
+						&ceremony_id_cloned,
+						request,
+						peers_map_clone,
+						enclave_lookup_cloned,
+						ceremony_registry_cloned,
+					);
+				});
 			});
 		},
 	}
 }
 
+// it will try to send request until it succeeds, the peer is removed from registry or ceremony is removed
+fn send_request<ECL, BKR>(
+	signer_id: SignerId,
+	ceremony_id: &CeremonyId,
+	request: RpcRequest,
+	peers_map: Arc<Mutex<HashMap<[u8; 32], DirectRpcClient>>>,
+	enclave_registry_lookup: Arc<ECL>,
+	ceremony_registry: Arc<RwLock<CeremonyRegistry<BKR>>>,
+) where
+	ECL: EnclaveRegistryLookup,
+	BKR: AccessKey<KeyType = SchnorrPair>,
+{
+	loop {
+		let client = peers_map.lock().unwrap().get(&signer_id).cloned();
+		if let Some(mut client) = client {
+			if let Err(e) = client.send(&request) {
+				error!("Could not send request to signer: {:?}, reason: {:?}", signer_id, e);
```

Comment:
Will it spam the logs?

Reply:
It can, currently 20 log entries per second at worst (20 threads).

We can either:
* change interval 
* have some kind of error deduplication - so we do not log the same error twice for request

In normal conditions this should not be logged at all.

Reply:
I've changed interval to 5 s.

Let's add deduplication/other impl preventing from logs being flooded in separate PR if we decide we need it